tests/NationalInstruments.Analyzers.UnitTests/ChainOfMethodsWithLambdasAnalyzerTests.cs

```diff
@@ -0,0 +1,648 @@
+            var distinctSelectedSoftware = <|selectedSoftwareContents.GroupBy(software => software.AliasName).Select(g => g.First()).Distinct();|>
+        }
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+}, // inside a method
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        Test(IEnumerable<ISoftwareContent> selectedSoftwareContents)
+        {
+            var distinctSelectedSoftware = <|selectedSoftwareContents.GroupBy(software => software.AliasName).Select(g => g.First())
+                .Distinct();|>
+        }
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+}, // inside a constructor
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        private IEnumerable<ISoftwareContent> selectedSoftwareContents = Enumerable.Empty<ISoftwareContent>();
+
+        IEnumerable<ISoftwareContent> DistinctSoftwareContents =>
+         <|selectedSoftwareContents.GroupBy(software => software.AliasName).Select(g => g.First())
+            .Distinct();|>
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+}, // inside a property getter
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(IEnumerable<ISoftwareContent> softwareContent)
+        {
+            softwareContent.Select(content => <|content.Children.GroupBy(software => software.AliasName).Select(g => g.First()).Distinct()|>);
+        }
+
+        private interface ISoftwareContent
+        {
+            IEnumerable<ISoftwareContent> Children { get; }
+
+            string AliasName { get; }
+        }
+    }
+}
+"
+} // inside an argument
+            };
+
+        [Fact]
+        public void NI1017_NoLambdas_NoDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(ISoftwareContent softwareContent)
+        {
+            softwareContent.Child().ToString().TrimEnd();
+            softwareContent.Child().AliasName.TrimEnd().Length.ToString();
+        }
+
+        private interface ISoftwareContent
+        {
+            ISoftwareContent Child();
+
+            string AliasName { get; }
+        }
+    }
+}
+");
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_OneLambda_NoDiagnostic()
```

Comment:
Could make this a theory with lines 265-268 each value so the actual differences are obvious (sub them into a source template).

Reply:
Done.